router-bridge/src/testdata/custom_cost_schema.graphql

```diff
@@ -0,0 +1,154 @@
+schema
+  @link(url: "https://specs.apollo.dev/link/v1.0")
+  @link(url: "https://specs.apollo.dev/join/v0.5", for: EXECUTION)
+  @link(
+    url: "https://specs.apollo.dev/cost/v0.1"
+    import: ["@cost", "@listSize"]
+  ) {
+  query: Query
+}
+
+directive @cost(
+  weight: Int!
+) on ARGUMENT_DEFINITION | ENUM | FIELD_DEFINITION | INPUT_FIELD_DEFINITION | OBJECT | SCALAR
+
+directive @cost__listSize(
```

Comment:
I'm confused why there's a directive definition for `@cost__listSize` here since `@listSize` was imported, the directive's name in schema should be `@listSize` instead of `@cost__listSize`.

Reply:
It's an artifact of composition when we have multiple directives defined in the same specification. I copied this over from other test cases to make sure it works. We can try to remove that extra definition later, but it's a composition change, so there's nothing I can do in this PR.

Reply:
@tninesling 
The `@link` spec generally assumes that directives in a spec have one name in a schema (and code that processes a schema using `@link` can and often do assume this). This is technically fine in the example schema given because the extra definition gets ignored, but applications of this definition may also be ignored (and that's usually not permissible).

You're right that this isn't a problem with this PR specifically, so I'll go ahead and approve, and release the new RC. I'll see if I can get the composition issue fixed tonight (if so, I'll cut a `2.9.0-beta.1` RC).